### DISCUSSION TITLE
Is required close watch?

### DISCUSSION THREAD

**Discussion by gustavorozolin** [isar/isar#355] [Help] [answered]
What are the best practices for work with watch?

Imagine this code:
   ```
Query<Student> query = this
        .isar!
        .students
        .filter()
        .nameContains(stringQuery, caseSensitive: false)
        .build();
    query.watch(initialReturn: true).listen((event) {
      this.setState(() {
        this.list = event;
      });
    }, onError: (o) {
      print('OnError '+o.toString());
    });
```
when I change the `queryString` it's required "close" or "stop listener" of old `Query` or Isar handler it's for me?

**Comment by simc** [comment, marked as answer]
Yes you are required to unsubscribe from the change stream if you no longer need the stream.

It works like every other Stream in Dart: `. listen()` returns a [`StreamSubscription`](https://api.dart.dev/stable/2.16.2/dart-async/StreamSubscription-class.html) which you must `.cancel()` when you're finished watching.
